core/src/main/scala-2.12/cats/data/ChainCompanionCompat.scala

```diff
@@ -0,0 +1,42 @@
+package cats.data
+
+import cats.data.Chain.{nil, one, Wrap}
+import cats.kernel.compat.scalaVersionSpecific.IterableOnce
+
+import scala.collection.immutable.{Seq => ImSeq}
+import scala.collection.mutable.{Seq => MutSeq}
+
+private[data] trait ChainCompanionCompat {
+
+  /**
+   * Creates a Chain from the specified sequence.
+   */
+  def fromSeq[A](s: Seq[A]): Chain[A] =
+    s match {
+      case mut: MutSeq[A] => fromMutableSeq(mut)
+      case imm: ImSeq[A]  => fromImmutableSeq(imm)
+    }
+
+  private def fromImmutableSeq[A](s: ImSeq[A]): Chain[A] = {
+    if (s.isEmpty) nil
+    else if (s.lengthCompare(1) == 0) one(s.head)
+    else Wrap(s)
+  }
```

Comment:
This implementation is identical in both Scala 2.12 and 2.13+ versions. I wonder – does it make sense to make it shared to avoid unnecessary duplication?

Reply:
Eh, I don't think it's worth the mental indirection when reading the code. It's just a few lines, it's fine.